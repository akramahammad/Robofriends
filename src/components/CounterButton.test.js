import {shallow} from 'enzyme'
import React from 'react'
import CounterButton from './CounterButton'



it("Rendering Counter button component",()=>{
    expect.assertions(1)
    const mockColor='red'
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
})

it('Correctly increments counter',()=>{
    const mockColor='red'
    const wrapper = shallow(<CounterButton color={mockColor} />)
    expect(wrapper.state().count).toEqual(1)
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state().count).toEqual(2)
    expect(wrapper.props().color).toEqual('red')
})
