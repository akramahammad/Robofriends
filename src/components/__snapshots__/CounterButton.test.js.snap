// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Counter button component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
